FieldOpt/Reservoir/grid/cell.cpp

```diff
@@ -52,24 +53,35 @@ namespace Reservoir {
         bool Cell::EnvelopsPoint(Eigen::Vector3d point) {
             bool point_inside = true;
             for (Face face : faces_) {
-                if ((point - face.corners[0]).dot(face.normal_vector) < 0)
+		double dot_prod = (point - face.corners[0]).dot(face.normal_vector);
+		if ( dot_prod < 0)
                     point_inside = false;
             }
             return point_inside;
         }
 
         void Cell::initializeFaces() {
+//            int face_indices_points[6][4] = {
+//                     {0, 2, 1, 3},
+//                     {4, 5, 6, 7},
+//                     {0, 4, 2, 6},
+//                     {1, 3, 5, 7},
+//                     {0, 1, 4, 5},
+//                     {2, 6, 3, 7}
+//            };
+//
+
             int face_indices_points[6][4] = {
-                    {0, 2, 1, 3},
-                    {4, 5, 6, 7},
-                    {0, 4, 2, 6},
-                    {1, 3, 5, 7},
-                    {0, 1, 4, 5},
-                    {2, 6, 3, 7}
+                    {2, 0, 3, 1},
```

Comment:
Hi Einar, 

(I'm not working on Mondays). Both indices are useful, I have new commits that will make sure the correct permutation is picked. It matters what type of grid the user is providing, namely if it right-handed of left-handed. For the unit tests the permutation that you have used is good but for the grids I had to work the one I included is working. It boils down to whether the z-axis is increasing toward the center of the Earth or towards the sky. In conclusion I will say to keep the permutation that is useful for the unit tests and I will commit in a new PR the code that makes sure the proper permutation is chosen by looking at the grid provided by the user. Cheers, Alin